sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/partition_processor.py

```diff
@@ -0,0 +1,47 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import List
+from abc import ABC, abstractmethod
+from enum import Enum
+from .checkpoint_manager import CheckpointManager
+
+from azure.eventhub import EventData
+
+
+class CloseReason(Enum):
+    SHUTDOWN = 0  # user call EventProcessor.stop()
+    OWNERSHIP_LOST = 1  # lose the ownership of a partition.
+    EVENTHUB_EXCEPTION = 2  # Exception happens during receiving events
+
+
+class PartitionProcessor(ABC):
+    def __init__(self, eventhub_name, consumer_group_name, partition_id, checkpoint_manager: CheckpointManager):
+        self._partition_id = partition_id
```

Comment:
I'm curious as to why these attributes are private?

Reply:
Changed PartitionProcessor to protocol. This is irrelevant any more